index.js

```diff
@@ -119,75 +121,101 @@ function createJsonRpcCapabilities ({ safeMethods = [], restrictedMethods = {},
       const { permissions } = domains[domain];
       return permissions;
     }
-    return {};
+    return [];
   };
 
-  /*
+  /**
    * Get the parent-most permission granting the requested domain's method permission.
+   * Follows the granter chain of the first matching permission found.
+   * 
+   * TODO:? Is the granter-chain following here sound? It could be that the same permission 
+   * has been granted by different domains with different root permission requests.
```

Comment:
I think a good solution for this problem is to allow the user to specify a permission ID to the `grantPermissionsMiddleware` (and `revokePermissionsMiddleware`). I added a `TODO` for this. May be worthy of its own issue?

On this note, I made it so that the `requestedPermissions` array (second element of `params` array in the `request` to the above mentioned middleware functions) expects objects rather than `methodName` strings, to enable such flexibility in the future. See e.g. [here](https://github.com/MetaMask/json-rpc-capabilities-middleware/blob/0faeee001c078550373108757db774e9097d4b10/index.js#L399) and [here](https://github.com/MetaMask/json-rpc-capabilities-middleware/blob/0faeee001c078550373108757db774e9097d4b10/test/grantPermissions.js#L16-L29).